src/Init/Control/Basic.lean

```diff
@@ -66,11 +66,100 @@ infixr:35 " <&&> " => andM
 @[macroInline] def notM {m : Type → Type v} [Applicative m] (x : m Bool) : m Bool :=
   not <$> x
 
+/-!
+
+# How `MonadControl` works
+
+There is a [tutorial by Alexis King](https://lexi-lambda.github.io/blog/2019/09/07/demystifying-monadbasecontrol/) that this docstring is based on.
+
+Suppose we have `foo : ∀ α, IO α → IO α` and `bar : StateT σ IO β` (ie, `bar : σ → IO (σ × β)`).
+We might want to 'map' `bar` by `foo`. Concretely we would write this as:
+
+```lean
+def mapped_foo : StateT σ m β → StateT σ m β := do
+  let s ← get
+  let (s', v) ← lift <| foo <| StateT.run bar s
+  set s'
+  pure v
```

Comment:
Could you please make sure that all (non-pseudo) examples are runnable as is? Especially since we might eventually want to test them automatically. There seem to be multiple issues in this code block.